R/nnmf_sparse.R

```diff
@@ -0,0 +1,267 @@
+#' Non-Negative Matrix Factorization Signal Extraction with lasso Penalization
+#'
+#' `step_nnmf_sparse()` creates a *specification* of a recipe step
+#'  that will convert numeric data into one or more non-negative
+#'  components.
+#'
+#' @inheritParams step_pca
+#' @inheritParams step_center
+#' @param num_comp The number of components to retain as new predictors. If
+#'  `num_comp` is greater than the number of columns or the number of possible
+#'  components, a smaller value will be used.
+#' @param penalty A non-negative number used as a penalization factor for the
+#' loadings. Values are usually between zero and one.
+#' @param options A list of options to `nmf()` in the RcppML package. That
+#'  package has a separate function `setRcppMLthreads()` that controls the
+#'  amount of internal parallelization. **Note** that the argument `A`, `k`,
+#'  `L1`, and `seed` should not be passed here.
+#' @param res A matrix of loadings is stored here, along with the names of the
+#'  original predictors, once this preprocessing step has been trained by
+#'  [prep.recipe()].
+#' @param prefix A character string that will be the prefix to the
+#'   resulting new variables. See notes below.
```

Comment:
We should carefully go through these arguments and make sure we aren't starting to repeat ourselves again here instead of inheriting. For example, `prefix` should inherit from `step_pca()`, as should `num_comp`.

Reply:
I have opened a new issue related to the `num_comp` argument since it is a broader problem. https://github.com/tidymodels/recipes/issues/884

Reply:
I changed `prefix`. We can handle `num,_comp` in a different PR